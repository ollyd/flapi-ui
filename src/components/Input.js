import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export default function Input(props) {
  const { value, setSearchInput } = props;

  return (
    <InputGroup>
      <StyledInput type="text" placeholder="Search" value={value} onChange={e => setSearchInput(e.target.value)} />
      <Bar />
    </InputGroup>
  );
}

const InputGroup = styled.div`
  position: fixed;
  top: 15px;
  z-index: 2;
  margin-left: auto;
  right: 15rem;

  @media (max-width: 960px) {
    right: 5rem;
  }

  @media (max-width: 600px) {
    right: 0.8rem;
  }
`;

const StyledInput = styled.input`
  ${({ theme }) => `
    font-size: 1.8rem;
    font-weight: 300;
    color: ${theme.palette.secondary};
    padding: 0.8rem 0.8rem 0.8rem 0.4rem;
    display: block;
    width: 30rem;
    border: none;
    border-bottom: 1px solid ${theme.palette.secondary};
    background-color: ${theme.palette.primary};
    margin-left: auto;
    margin-top: 0.8rem;

    &:focus { outline: none; }

    @media (max-width: 600px) {
      width: 20rem;
    }
  `}
`;

const Bar = styled.span`
  ${({ theme }) => `
    position: relative;
    display: block;
    width: 30rem;

    &:before, &:after {
      content: '';
      height: 0.2rem;
      width: 0;
      bottom: 0.1rem;
      position: absolute;
      background: ${theme.palette.secondary};
      transition: 0.2s ease all;
    }
  `}
`;

Input.propTypes = {
  value: PropTypes.string,
  setSearchInput: PropTypes.func.isRequired,
};

Input.defaultProps = {
  value: '',
};
