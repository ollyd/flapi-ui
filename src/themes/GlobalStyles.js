import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  ${({ theme }) => `
    html {
      font-family: 'Roboto', sans-serif;
      font-size: 62.5%;
    }

    body {
      background-color: ${theme.palette.contrast};
      color: ${theme.text.color.primary};
      font-size: 1.6rem;
      line-height: 1.5;
      margin: 0;
      padding: 8rem 15rem;

      @media (max-width: 960px) {
        padding: 8rem 5rem;
      }

      @media (max-width: 600px) {
        padding: 8rem 0.8rem;
      }

      &.no-scroll {
        overflow: hidden;
      }
    }

    html,
    body,
    #root {
      height: 100%;
    }

    * {
      box-sizing: border-box;
    }

    a {
      color: ${theme.text.color.primary};
      text-decoration: none;
    }
  `}
`;
