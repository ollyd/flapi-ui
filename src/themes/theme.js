export default {
  text: {
    color: {
      primary: 'rgba(0,0,0,0.67)',
      contrast: '#fff',
    },
  },
  palette: {
    primary: '#34495e',
    secondary: '#1abc9c',
    contrast: '#ecf0f1',
  },
  border: '1px solid #d2d2d2',
};
