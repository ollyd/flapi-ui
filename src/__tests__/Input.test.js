/* eslint no-undef: "off" */
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import Input from '../components/Input';
import theme from '../themes/theme';
import GlobalStyles from '../themes/GlobalStyles';

afterEach(cleanup);

const waitForNextTick = () => new Promise(resolve => setTimeout(resolve), 501);

const setup = (setSearchInput, value) => {
  const { container, rerender, getByPlaceholderText } = render(
    <ThemeProvider theme={theme}>
      <>
        <GlobalStyles />
        <Input setSearchInput={setSearchInput} value={value} />
      </>
    </ThemeProvider>,
  );

  const input = getByPlaceholderText('Search');

  return {
    input,
    container,
    rerender,
  };
};

describe('<Input>: Style snapshot tests', () => {
  it('should have correct styles', () => {
    const { container } = setup(jest.fn(), 'cats');
    expect(container).toMatchSnapshot();
  });
});

describe('<Input>: Unit tests', () => {
  it('should have the correct value after onChange', async () => {
    let value = 'cats';
    const setSearchInput = jest.fn((ev) => {
      value = ev;
    });

    const { input } = setup(setSearchInput, value);
    expect(value).toBe('cats');

    fireEvent.change(input, { target: { value: 'dogs' } });

    await waitForNextTick();

    expect(setSearchInput).toHaveBeenCalledTimes(1);
    expect(value).toBe('dogs');
  });
});
