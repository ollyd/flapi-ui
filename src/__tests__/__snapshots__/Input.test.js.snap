// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input>: Style snapshot tests should have correct styles 1`] = `
.c0 {
  position: fixed;
  top: 15px;
  z-index: 2;
  margin-left: auto;
  right: 15rem;
}

.c1 {
  font-size: 1.8rem;
  font-weight: 300;
  color: #1abc9c;
  padding: 0.8rem 0.8rem 0.8rem 0.4rem;
  display: block;
  width: 30rem;
  border: none;
  border-bottom: 1px solid #1abc9c;
  background-color: #34495e;
  margin-left: auto;
  margin-top: 0.8rem;
}

.c1:focus {
  outline: none;
}

.c2 {
  position: relative;
  display: block;
  width: 30rem;
}

.c2:before,
.c2:after {
  content: '';
  height: 0.2rem;
  width: 0;
  bottom: 0.1rem;
  position: absolute;
  background: #1abc9c;
  -webkit-transition: 0.2s ease all;
  transition: 0.2s ease all;
}

@media (max-width:960px) {
  .c0 {
    right: 5rem;
  }
}

@media (max-width:600px) {
  .c0 {
    right: 0.8rem;
  }
}

@media (max-width:600px) {
  .c1 {
    width: 20rem;
  }
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search"
      type="text"
      value="cats"
    />
    <span
      class="c2"
    />
  </div>
</div>
`;
